### DISCUSSION TITLE
Manual linux build: debug only?

### DISCUSSION THREAD

**Discussion by dackelblut** [Chatterino/chatterino2#4290] [Q&A] [answered]
Hi,
I tried to build v2.4.0 manually on linux with cmake and with qtcreator and I got a debug build both times. qtcreator warned during import/opening of CMakeLists.txt of a missing BUILD_TYPE variable but showed Release (in the build type selection menu) nonetheless. What do I have to adjust to get a release build?

**Comment by Felanbird** [comment]
![image](https://user-images.githubusercontent.com/41973452/211170413-abf6f653-0584-4e32-a375-4ab994eefecc.png)
Select this

**Comment by dackelblut** [comment]
Thanks for your help, but no, I don't have the build selection column. I know qtcreator and that selection menu from other projects and this is missing in this case. Release is shown left and then there's only the Run column on the right.

**Comment by pajlada** [comment, marked as answer]
When first opening the CMakeLists.txt from chatterino2 in Qt Creator you will be presented with the options to configure the project, and select what kits to use.
Kits are what decides build flags. I am presented with options to enable Debug, Release, Release with Debug Information, Profile, and Minimum Size Release builds
![image](https://user-images.githubusercontent.com/962989/211173008-7170d576-7f71-476b-8772-6c3e4d90e30d.png)

Once I've configured the project, all the selected kits are available in the bottom right
![image](https://user-images.githubusercontent.com/962989/211173018-5d9f413e-659e-4f44-9ba1-fcf376a71280.png)

After a project is configured, the kits are stored in a CMakeLists.txt.user file in the same directory as CMakeLists.txt, so if I close the project, delete the CMakeLists.txt.user file and open the project again I'm presented with the Configure Project screen again.

To add kits, you can click the Projects list on the left and click the "Add" dropdown
![image](https://user-images.githubusercontent.com/962989/211173079-7d8e13fc-7d19-426e-847e-42d3e8daf703.png)

**Comment by dackelblut** [comment]
Thank you for your detailed guide, @pajlada, but no luck so far, the problem persists. Even with a fresh restart, a newly cloned copy, I get the same output. If I run a build the manual way with _cmake .. -DCMAKE_BUILD_TYPE=Release_ the build does not have DEBUG in the title but still spams the console with verbose fetching of links/emotes etc.  and diagnostic lines.

**Reply by pajlada** [reply]
Debug/Release builds don't control log output, to control the log output you will need to use the `QT_LOGGING_RULES` and `QT_MESSAGE_PATTERN` environment variables as documented [here](https://wiki.chatterino.com/Environment%20Variables/#qt_logging_rules)

**Comment by iupsal** [comment]
I managed to built it after installing libsecret-1 and qtkeychain packages via my package manager (on Linux Mint). I think there may be missing dependencies.

**Reply by iupsal** [reply]
Oh, also libboost-all-dev and its dependencies. All I can say with confidence is that it wasn't working before I got these and it worked after, but I wasn't so careful with testing.
